larray/inout/excel.py

```diff
@@ -320,20 +320,63 @@ def __setitem__(self, key, value):
 
         @property
         def shape(self):
-            # include top-left empty rows/columns
+            """
+            shape of sheet including top-left empty rows/columns but excluding bottom-right ones.
+            """
             from xlwings.constants import Direction as xldir
-            used = self.xw_sheet.api.UsedRange
-            last_row = used.Row + used.Rows.Count
-            last_col = used.Column + used.Columns.Count
-            for last_used_row in range(last_row, 0, -1):
-                left_cell = used.Cells(last_used_row, last_col + 1).End(xldir.xlToLeft)
-                if left_cell.Column > 1 or left_cell.Value is not None:
-                    break
-            for last_used_col in range(last_col, 0, -1):
-                up_cell = used.Cells(last_row + 1, last_used_col).End(xldir.xlUp)
-                if up_cell.Row > 1 or up_cell.Value is not None:
-                    break
-            return (last_used_row, last_used_col)
+
+            sheet = self.xw_sheet.api
```

Comment:
api = self.xw_sheet.api ?

Reply:
sheet is more descriptive. It's a "raw" win32com sheet object.